### PR TITLE
Add items table to index page

### PR DESCRIPTION
**Pull request by Middlebrooks314**
**Summary**

The `add-items-index` branch is adding the feature for the user to be able to go to the Gilded Rose LLC store webpage and view a table of all the items in the store. 

itemsTable

**Follow up work**

The next feature will build onto this feature by adding the ability to view the details of an individual item.

**Questions**

- This conditional rendering statement ended up fairly verbose with the HTML. I tried to extract the code into a variable or a method, and I was unable to get it to work. I finally retired to just get it working first. I would love feedback on this. 

itemsIndex

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -28,14 +40,19 @@ class App extends Component {
...
+    const { isLoaded, error, text, items } = this.state
     if (error) {
       return <div data-testid="error-message">{error}</div>
     } else if (!isLoaded) {
       return <div data-testid="loading-message">Loading...</div>
     } else {
       return (
+        <div>
           <Welcome welcomeString={text}/>
+          {this.state.items && (
```
Comment (mbdebbeler) [inline src/App.js:52]: **praise**: good job handling the edge case where we have no items!

File: src/App.test.js [hunk 2]
```diff
@@ -30,4 +30,11 @@ describe('App', () => {
       screen.getByTestId('error-message')
     ).toBeInTheDocument()
   })
+  it('renders an items table', async () => {
```
Comment (mbdebbeler) [inline src/App.test.js:33]: **suggestion**: Add a test for the sad path here, since you handled it! Ex: this test becomes "When there are items, renders an items table." and the new test is "When there are no items, does not render an items table."

File: src/components/ItemsIndex/ItemsIndex.test.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import '@testing-library/jest-dom/extend-expect'
+import App from '../../App'
+import MockApiHandler from '../../services/__mocks__/mockApiHandler'
+
+describe('ItemsIndex', () => {
```
Comment (mbdebbeler) [inline src/components/ItemsIndex/ItemsIndex.test.js:7]: **praise:** such comprehensive cases!

File: src/services/__mocks__/mockApiHandler.js [hunk 4]
```diff
@@ -1,11 +1,51 @@
...
+    quality: 16
+  }
+]
+}
 class MockApiHandler {
-  constructor (text, error) {
+  constructor (text, error, items) {
     this.text = text
     this.error = error
+    this.items = items || storeItems
```
Comment (mbdebbeler) [inline src/services/__mocks__/mockApiHandler.js:40]: **praise**: love this default argument!

**Review by mbdebbeler — APPROVED** [review]

File: src/services/__mocks__/mockApiHandler.js [hunk 5]
```diff
@@ -1,11 +1,51 @@
+const storeItems = {
```
Comment (cmvandrevala) [inline src/services/__mocks__/mockApiHandler.js:1]: I like this a lot. Nice

**Review by cmvandrevala — APPROVED** [review]
